web3/main.py

```diff
@@ -198,14 +198,14 @@ def middleware_onion(self) -> MiddlewareOnion:
     def to_bytes(
         primitive: Primitives = None, hexstr: HexStr = None, text: str = None
     ) -> bytes:
-        return to_bytes(primitive, hexstr, text)
+        return cast(bytes, to_bytes(primitive, hexstr, text))
```

Comment:
These don't seem, and shouldn't be, necessary locally. If there's any issue here we should address it some other way. Casting everything that `to_bytes`, `to_int`, etc, returns will not be an ideal way to resolve things across the library.

However, taking these out does not give me lint issues locally... so not sure if they are needed in the first place?

Reply:
I was attempting to fix `no-any-return` `mypy` errors and started by `cast`ing them before I realized that it was a much bigger problem. If you remove the cast and set `mypy` `warn_return_any = true`, it'll come up.